Subject: add some delays to avoid endless resets
From: olh@suse.de
References: 159235 - LTC22345
status 2.6.19

the hald media changed polling does really confuse things.
noone knows why the delays are needed, but they give us access to
the CD

---
 drivers/ata/pata_sl82c105.c |    3 +++
 1 file changed, 3 insertions(+)

Index: linux-2.6.16-SLES10_SP1_BRANCH/drivers/ata/pata_sl82c105.c
===================================================================
--- linux-2.6.16-SLES10_SP1_BRANCH.orig/drivers/ata/pata_sl82c105.c
+++ linux-2.6.16-SLES10_SP1_BRANCH/drivers/ata/pata_sl82c105.c
@@ -187,7 +187,9 @@ static void sl82c105_bmdma_start(struct 
 {
 	struct ata_port *ap = qc->ap;
 
+	udelay(100);
 	sl82c105_reset_engine(ap);
+	udelay(100);
 
 	/* Set the clocks for DMA */
 	sl82c105_configure_dmamode(ap, qc->dev);
@@ -216,6 +218,7 @@ static void sl82c105_bmdma_stop(struct a
 
 	ata_bmdma_stop(qc);
 	sl82c105_reset_engine(ap);
+	udelay(100);
 
 	/* This will redo the initial setup of the DMA device to matching
 	   PIO timings */
